include/deal.II/lac/sparse_matrix_tools.h

```diff
@@ -151,6 +151,9 @@ namespace SparseMatrixTools
       (void)comm;
       return {0, value};
 #  else
+      if (comm == MPI_COMM_SELF)
+        return {0, value}; // serial triangulation
```

Comment:
I don't think this is the right tests. You want to check whether the communicator has one process attached. It doesn't have to be `MPI_COMM_SELF` -- it could be `MPI_COMM_WORLD` or a sub-communicator, as long as it's only got one process. I think it would be better to test `Utilities::MPI::n_mpi_processes(comm)==1` here.

Separately, the comment is not useful. There is no triangulation available in this function.